src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Mappers/AssemblyMapperTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Microsoft.DotNet.ApiCompatibility.Rules;
+using Moq;
```

Comment:
Is using `Moq` preferable over implementing our own `IAssemblyMapper` that keeps track of the settings, size, etc.?
Likewise for the other interfaces.

Reply:
Using [Moq](https://github.com/moq/moq4#moq) avoids implementing interfaces yourself when you just need a simple (or even a complex) implementation. Without Moq, dependencies would need to be manually implemented.

In code above, you just want to mock the dependency without specifying any explicit behavior, because the dependency isn't used as part of the AssemblyMapper. You just want to verify that what is passed into the ctor is correctly set as a property.

For other units (not part of this PR), the dependencies are actually used, i.e. to retrieve members from a type and such mocks need to have some behavior attached which is also straightforward with Moq. I.e. https://github.com/dotnet/sdk/blob/fb2651af5fec4a910ddcd9b8007000debceaab18/src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Runner/ApiCompatRunnerTests.cs#L20-L26